File: generate-text/summaries-chapters-highlights.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))

result = client.generate.summarize(
    video_id="66ebe4bfd4f452b062c63e3a",
    prompt="Be concise",
    temperature=0.7,
    type="summary"
)

print(f"Result ID: {result.id}")

if result.summary is not None:
    print(f"Summary: {result.summary}")

if result.chapters is not None:
    print("Chapters:")
    for chapter in result.chapters:
        print(f"  Chapter {chapter.chapter_number}:")
        print(f"    Start: {chapter.start}")
        print(f"    End: {chapter.end}")
        print(f"    Title: {chapter.chapter_title}")
        print(f"    Summary: {chapter.chapter_summary}")

if result.highlights is not None:
    print("Highlights:")
    for highlight in result.highlights:
        print(f"  Start: {highlight.start}")
        print(f"  End: {highlight.end}")
        print(f"  Highlight: {highlight.highlight}")

File: generate-text/summaries-chapters-highlights.js
import { TwelveLabs } from "twelvelabs-js";

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY });

  const result = await client.generate.summarize(
    "66ebe4bfd4f452b062c63e3a",
    "highlight",
    "Be concise",
    0.7,
  );

  console.log(`Result ID: ${result.id}`);

  if (result.summary !== undefined) {
    console.log(`Summary: ${result.summary}`);
  }

  if (result.chapters !== undefined && result.chapters.length > 0) {
    console.log("Chapters:");
    result.chapters.forEach((chapter) => {
      console.log(`  Chapter ${chapter.chapterNumber}:`);
      console.log(`    Start: ${chapter.start}`);
      console.log(`    End: ${chapter.end}`);
      console.log(`    Title: ${chapter.chapterTitle}`);
      console.log(`    Summary: ${chapter.chapterSummary}`);
    });
  }

  if (result.highlights !== undefined && result.highlights.length > 0) {
    console.log("Highlights:");
    result.highlights.forEach((highlight) => {
      console.log(`  Start: ${highlight.start}`);
      console.log(`  End: ${highlight.end}`);
      console.log(`  Highlight: ${highlight.highlight}`);
    });
  }
})();
